common/oatbox/cache/SetupFileCache.php

```diff
@@ -0,0 +1,54 @@
+<?php
+/**
+ * This program is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU General Public License
+ * as published by the Free Software Foundation; under version 2
+ * of the License (non-upgradable).
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program; if not, write to the Free Software
+ * Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+ *
+ * Copyright (c) 2022 (original work) Open Assessment Technologies SA ;
+ */
+
+declare(strict_types=1);
+
+namespace oat\oatbox\cache;
+
+use oat\generis\persistence\PersistenceManager;
+use oat\oatbox\service\ConfigurableService;
+
+class SetupFileCache extends ConfigurableService
+{
+    public const PERSISTENCE = 'cache';
+
+    public function createDirectory($cachePath): bool
+    {
+        if (is_dir($cachePath) && is_writable($cachePath)) {
+            return true;
+        }
+
+        return @mkdir($cachePath, 0700, true);
+    }
+
+    public function createPersistence(): void
```

Comment:
Are you sure this is the place to have this method? This is a `SetupFileCache` script.

Reply:
Thats how tao cache works, a KeyValueCache with `phpfile` driven `cache` persistence, the folder itself isn't a prerequisite, since setting a value creates the necessary folder structure. I believe this part can live in `SetupFileCache` 🤔

Reply:
Got it. It makes sense in this case. :) 

Could I do a small optional suggestion? use the prefix `setup` in the public methods to be consistent with the class name. Example: `setupPersistence` and `setupStorage` (instead of `createDirectory`).

Or ir possible, have just a single method `setup` that does all the work, but this is optional suggestion :)